File: 2022-2023/sourcecode/lab/lab12/ModulesBasicsJava/app/src/main/java/pl/edu/uwr/pum/modulesbasicsjava/di/ComputerComponent.java
package pl.edu.uwr.pum.modulesbasicsjava.di;

import dagger.Component;
import pl.edu.uwr.pum.modulesbasicsjava.MainActivity;
import pl.edu.uwr.pum.modulesbasicsjava.di.modules.ComputerModule;

@Component(modules = ComputerModule.class)
public interface ComputerComponent {
    void inject(MainActivity mainActivity);
}

File: 2022-2023/sourcecode/lab/lab12/ModulesBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/modulesbasicskotlin/di/ComputerComponent.kt
package pl.edu.uwr.pum.modulesbasicskotlin.di

import dagger.Component
import pl.edu.uwr.pum.modulesbasicskotlin.MainActivity
import pl.edu.uwr.pum.modulesbasicskotlin.di.modules.ComputerModule

@Component(modules = [ComputerModule::class])
interface ComputerComponent {
    fun inject(activity: MainActivity)
}